store/postgres/src/deployment_store.rs

```diff
@@ -1362,14 +1370,63 @@ impl DeploymentStore {
                 // changes that might need to be reverted
                 Layout::revert_metadata(conn, &site, block)?;
 
-                deployment::update_entity_count(conn, site.as_ref(), count)?;
+                if truncate {
+                    deployment::set_entity_count(conn, site.as_ref(), layout.count_query.as_str())?;
+                } else {
+                    deployment::update_entity_count(conn, site.as_ref(), count)?;
+                }
+
                 Ok(event)
             })
         })?;
 
         Ok(event)
     }
 
+    fn rewind_with_conn(
```

Comment:
Since `rewind_with_conn` and `truncate_with_conn` are so simple, they could be inlined into their callers.

Reply:
I did so since these both give two very different results and thought it'd make sense to not use `rewind_or_truncate_with_conn` that has a boolean flag `truncate` everywhere but instead separate it to two functions. 

Will inline it and use `rewind_or_truncate_with_conn` instead.